recipe detail unavailable  customization (#28)
miamtech/mealz-documentation: docs/android/features/_shared/_custom-views/recipeDetail/RecipeDetailUnavailable.mdx

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> ### unavailable
> 
> Unavailable has two sub-templates : 
> - **header** : The line before the list of products
> - **product** : product it self

Reviewer: ```suggestion
### Unavailable

Unavailable has two sub-templates : 
- **header** : The toggle button that programmatically displays or hides the list of products
- **product** : A single product in the list
```

```kotlin
unavailable {
    header  {
        view = MyCustomRecipeDetailUnavailableHeader()
    }
    product {
        view = MyCustomRecipeDetailUnavailableProduct()
    }
}
```

> #### header
> 
> To create your own recipe detail oftend deleted product template you have to create class that implement `UnavailableProductsHeader`

Reviewer: ```suggestion
#### Header

To create your own recipe detail unavailable product template you create a class that implements `UnavailableProductsHeader`
```

         val rotationState by animateFloatAsState(
            targetValue = if (params.isOpen) 90f else 0f, label = "Icon rotation state "
        )

        Row(
            verticalAlignment = Alignment.CenterVertically,
            modifier = Modifier
                .fillMaxWidth()
                .padding(start = 16.dp)
        ) {
            Text(
                text = params.title,
                modifier = Modifier.weight(6f),
                maxLines = 1,
                fontWeight = if (params.isOpen) FontWeight.Bold else FontWeight.Normal
            )
            IconButton(modifier = Modifier
                .weight(1f)
                .rotate(rotationState), onClick = { params.toggle() }) {
                Image(
                    painter = painterResource(ai.mealz.sdk.ressource.Image.toggleCaret),
                    contentDescription = "open icon",
                    modifier = Modifier
                )
            }
        }
    }
}
```
</TabItem>
</Tabs>

with 

```kotlin
data class UnavailableProductsHeaderParameters(
    val title: String,
    val isOpen: Boolean,
    val toggle: () -> Unit
)
```

> #### product
> 
> To create your own recipe detail unavailable product template you have to create class that implement `UnavailableProduct`

Reviewer: ```suggestion
#### Product

To create your own recipe detail unavailable product template you create a class that implements `UnavailableProduct`
```
